Calculate the Required Sample Size:
Imagine the current checkout process has a conversion rate of 5%, and the bakery’s team is confident the new process will boost this to 7%.
Use Python to calculate the required sample size per group (current vs. new checkout process) with an effect size of 0.2, a significance level of 0.05, and a desired power of 0.8.

In [1]:
import statsmodels.stats.power as power

effect_size = 0.2
alpha = 0.05
power_value = 0.8

sample_size = power.tt_ind_solve_power(
    effect_size=effect_size,
    alpha=alpha,
    power=power_value,
    alternative='two-sided'
)

print(f"Required sample size per group: {round(sample_size)}")

Required sample size per group: 393


# **the bakery needs about 393 customers to use the current checkout process and 393 customers to use the new checkout process to confidently see if there's a real difference**

Analyze the Impact of Effect Size:

The bakery’s head chef, always aiming for perfection, wonders what would happen if the effect size were different. Calculate the required sample size for effect sizes of 0.1, 0.2, 0.3, and 0.4.
Explain how the sample size requirements change as the effect size changes. Why does this happen?



In [2]:
effect_sizes = [0.1, 0.2, 0.3, 0.4]
alpha = 0.05
power_value = 0.8

for effect_size in effect_sizes:
    sample_size = power.tt_ind_solve_power(
        effect_size=effect_size,
        alpha=alpha,
        power=power_value,
        alternative='two-sided'
    )
    print(f"Effect size: {effect_size}, Required sample size per group: {round(sample_size)}")

Effect size: 0.1, Required sample size per group: 1571
Effect size: 0.2, Required sample size per group: 393
Effect size: 0.3, Required sample size per group: 175
Effect size: 0.4, Required sample size per group: 99


# **The smaller the effect size (a smaller improvement), the more customers you need to see the difference. The larger the effect size (a bigger improvement), the fewer customers you need. This happens because bigger changes are easier to see. Smaller changes need more data to be sure they are real.**

Explain the Relationship:

Imagine you’re explaining this to the bakery’s team in a fun, easy-to-understand way. Why is it so important to balance effect size and sample size when planning an A/B test? Help them understand how this ensures they’re not wasting time or resources and how it helps them confidently make decisions that could increase their sweet sales.

# **Lets suppose we're testing a new sprinkle on our cupcakes. If the new sprinkle makes a HUGE difference, everyone buys them. We don't need to sell many to know it's a hit. But, if the new sprinkle makes a VERY small difference, maybe just a tiny bit more sparkly, we need to sell a LOT of cupcakes to see if people really like it more. Effect size is like how big the difference is. Sample size is like how many cupcakes we sell. If we test too few cupcakes, we might miss a good sprinkle (or think a bad sprinkle is good). If we test way too many, we're wasting sprinkles and time. Why it Matters: it saves time and money, it leads to reliable decisions, it reduces the chance of making changes that hurt sales.**